zappa/core.py

```diff
@@ -2930,11 +2940,20 @@ def get_scheduled_event_name(self, event, function, lambda_name, index=0):
             # prefix all entries bar the first with the index
             # Related: https://github.com/Miserlou/Zappa/pull/1051
             name = "{}-{}".format(index, name)
+
+        # https://github.com/zappa/Zappa/issues/1036
+        # Error because the name cannot be obtained if the function name is longer than 64 characters
+        if len(name) > max_name_length:
+            raise EnvironmentError(
+                "Too long schedule function name. Please shortening the schedule function name:"
```

Comment:
`function` is checked above, and will raise an error already.

Do we need to check both the `function`, and the `event['name']`?
It seems that we could just check the resulting `name` only.

Reply:
"-" cannot be used for "function", but can be used for "name" Therefore, we have separated these two.

Reply:
Can you update the error message for clarity?

> Generated scheduled event name is too long, shorten the function name!

Reply:
And include the resulting `name` in the error message.